frame/ranked-collective/src/lib.rs

```diff
@@ -382,12 +382,19 @@ pub mod pallet {
 		type RuntimeEvent: From<Event<Self, I>>
 			+ IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
-		/// The origin required to add or promote a mmember. The success value indicates the
+		/// The origin required to add a member.
+		type AddOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = ()>;
+
+		/// The origin required to remove a member. The success value indicates the
+		/// maximum rank *from which* the removal may be.
+		type RemoveOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Rank>;
+
+		/// The origin required to promote a member. The success value indicates the
 		/// maximum rank *to which* the promotion may be.
 		type PromoteOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Rank>;
```

Comment:
does it improve flexibility?
`PromoteOrigin` returns the `Rank`, if it returns 0, it can only `add_member`.
Same with `DemoteOrigin`.

Reply:
The `AddOrigin` allows a chain to use a different (from the `PromoteOrigin`) authority to add members. E.g., entering the ranked collective could be managed separately from promoting inside it. The same with the `RemoveOrigin`.

Reply:
you can do the same with only `PromoteOrigin` since it returns `Rank`

Reply:
In current impl where only `PromoteOrigin` exists, one could add a dedicated origin with a 0 rank to the `PromoteOrigin` to add new members. However, anyone who can promote can also add a new member in this case, not only our dedicated origin. So we can't express the setting where one can promote an existing member but can't add a new one.